src/Contracts/CancelTasksQuery.php

```diff
@@ -9,4 +9,23 @@
 class CancelTasksQuery
 {
     use TasksQueryTrait;
+
+    private array $canceledBy;
+
+    public function setCanceledBy(array $canceledBy)
+    {
+        $this->canceledBy = $canceledBy;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        return array_filter(
+            array_merge(
+                $this->baseArray(),
+                ['canceledBy' => $this->formatArray($this->canceledBy ?? null)]
+            ), function ($item) { return null != $item || is_numeric($item); }
+        );
+    }
```

Comment:
I thought that this method was removed from `TasksQueryTrait` specifically because it was not used in `CancelTasksQuery` no?

Reply:
Not really. This method is required for every `Query` class. The problem is not correct to access stuff from the `Trait`.  So now, the idea is that the Trait generate most of the content (the shared part), and I just add the missing parts, like `canceledBy` in this case.

Reply:
Yes, but technically you can't cancel a task that has already been canceled so if there is no point in filtering them via `canceledBy` as they have already been canceled yet no?
Or I missed something